src/flamegraph/mod.rs

```diff
@@ -429,6 +429,7 @@ where
             }
             stack.push(' ');
             stack.push_str(&line[samples_idx..]);
+            let stack = stack.trim();
```

Comment:
```suggestion
            // Trim to handle the case where functions names internally contain `;`.
            // This can happen, for example, with types like `[u8; 8]` in Rust.
            // See https://github.com/jonhoo/inferno/pull/338.
            let stack = stack.trim();
```